README.md

```diff
@@ -331,6 +331,51 @@ export default Modifier.extend({
 
 Whenever possible, it is recommended that you use the default "modern" API instead of the classic API.
 
+## TypeScript
+
+Using the "modern" native class API, you can use `.ts` instead of `.js` and it'll just work, as long as you do runtime checks to narrow the types of your args when you access them. 
+
+```ts
+// app/modifiers/scroll-position.ts
+import Modifier from 'ember-class-based-modifier';
+
+export default class ScrollPositionModifier extends Modifier {
+  // ...
+}
+```
+
+But to avoid writing runtime checks, you can extend `Modifier` with your own args, similar to the way you would define your args for a Glimmer Component. 
+
+```ts
+// app/modifiers/scroll-position.ts
+import Modifier from 'ember-class-based-modifier';
+
+interface ScrollPositionModifierArgs {
+  positional: number[],
+  // this modifier doesn't accept named args, but this is included as an example:
```

Comment:
We can use the `track-click` example, or make up a different one 😄

Reply:
I thought about using the `track-click` example since it has both positional and named args. But the named args seem deliberately flexible in that example to allow anything to be passed along to `trackEvent`. Keeping that flexibility would prevent you from narrowing down the types of the named args.

Reply:
what if we add a `relative: boolean` argument to this example which changes whether we interpret the number given as absolute scroll position or relative to its current position?

Reply:
I like that idea. Can I add it to the other `scroll-position` examples as well?

Reply:
This is what I think you mean:
```ts
didReceiveArguments() {
  if(this.args.named.relative) {
    this.element.scrollTop += this.scrollPosition;
  } else {
    this.element.scrollTop = this.scrollPosition;
  }
}
```

Reply:
Yes! I would make a named getter for it but seems good to me!